lib/sprite_batch.dart

```diff
@@ -0,0 +1,82 @@
+import 'dart:ui';
+
+import 'package:flutter/foundation.dart';
+
+import 'flame.dart';
+
+class SpriteBatch {
+  Image atlas;
+  List<Rect> rects = [];
+  List<RSTransform> transforms = [];
+  List<Color> colors = [];
+
+  static const defaultBlendMode = BlendMode.srcOver;
+  static const defaultCullRect = null;
+  static final defaultPaint = Paint();
+  static final defaultTransform = RSTransform(1, 0, 0, 0);
+  static const defaultColor = const Color(0x00000000); // transparent
+
+  SpriteBatch(String fileName) {
+    Flame.images.load(fileName).then((image) => atlas = image);
+  }
+
+  SpriteBatch.fromImage(this.atlas);
+
+  int get width => atlas.width;
```

Comment:
Just to make sure, I would verify for null here

Example
```dart
  int get width => atlas?.width;
```

Same thing for `height` and `size`

Reply:
do `atlas?.height ?? 0` or `atlas?.height` and return `null`

for size do:
```
  Size get size => atlas == null ? null : Size(width.toDouble(), height.toDouble());
```
or return size with 0 values ?

btw: i think i would prefer to pass an image to the constructor, which is not null, so we dont have to do async load stuff and null checks there. assets could be loaded outside and prepared beforehand

Reply:
I think we should just return null, not 0 as default, just to avoid a NPE inside the method.

Reply:
In the latest commit you dont have to check for null